react-client/src/pages/plan/research-outputs/researchoutputs-new.js

```diff
@@ -0,0 +1,533 @@
+import {
+    useEffect,
+    useState,
+    Fragment
+} from "react";
+import { useNavigate, useParams } from "react-router-dom";
+
+import {
+    DmpModel,
+    DataObject,
+    DataRepository,
+    getDmp,
+    saveDmp,
+    getOutputTypes
+} from "../../../models.js";
+import { fileSizeUnits } from "../../../utils.js";
+
+import TypeAhead from "../../../components/TypeAhead.jsx";
+import TextInput from "../../../components/text-input/textInput.js";
+import TextArea from "../../../components/textarea/textArea.js";
+import Select from "../../../components/select/select.js";
+import RadioButton from "../../../components/radio/radio";
+import SizeInput from "../../../components/size-input.js";
+import Spinner from "../../../components/spinner";
+
+import "./researchoutputs.scss";
+
+
+function ResearchOutputs() {
+    let navigate = useNavigate();
+    let sizeUnits = {};
+    fileSizeUnits.forEach(v => sizeUnits[v] = v);
+
+    const { dmpId } = useParams();
+    const [dmp, setDmp] = useState(null);
+    const [outputTypes, setOutputTypes] = useState({});
+    const [editIndex, setEditIndex] = useState(null);
+    const [dataObj, setDataObj] = useState(new DataObject({}));
+    const [working, setWorking] = useState(false);
+    const [otherField, setOtherField] = useState(false);
+
+
+    useEffect(() => {
+        getDmp(dmpId).then(initial => {
+            setDmp(initial);
+        });
+
+        getOutputTypes().then((data) => {
+            setOutputTypes(data);
+        });
+    }, [dmpId]);
+
+
+    function handleChange(ev) {
+        const { name, value } = ev.target;
+
+        switch (name) {
+            case "data_type":
+                var newObj = new DataObject(dataObj.getData());
+                newObj.type = value;
+                setDataObj(newObj);
+                break;
+
+            case "personal_info":
+                var newObj = new DataObject(dataObj.getData());
+                newObj.personal = value;
+                setDataObj(newObj);
+                break;
+
+            case "sensitive_data":
+                var newObj = new DataObject(dataObj.getData());
+                newObj.sensitive = value;
+                setDataObj(newObj);
+                break;
+
+            case "repository":
+                var newObj = new DataObject(dataObj.getData());
+                newObj.repository = new DataRepository(dataObj.repository.getData());
+
+                if (ev.data) {
+                    newObj.repository = new DataRepository(dataObj.repository.getData());
```

Comment:
Isn't this one redundant? You're initializing just above `if (ev.data)` here.

Reply:
Hi @briri , I actually deleted the "researchoutputs-new.js", since I updated researchoutputs.js with the new code.

Reply:
Oh, @briri this is this is the branch that was already merged. The new PR is here: https://github.com/CDLUC3/dmptool/pull/593